addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1261,8 +1262,17 @@ const Wysiwyg = Widget.extend({
                         anchorOffset = focusOffset = index;
                     }
                 } else {
-                    anchorNode = link;
-                    focusNode = link;
+                    const selectionDirection = getCursorDirection(selection.anchorNode, 0, selection.focusNode, 0);
+                    if (
+                        closestElement(selection.anchorNode, 'a') === link &&
+                        closestElement(selection.focusNode, 'a') === link
+                    ) {
+                        [anchorNode, focusNode] = selectionDirection
```

Comment:
I know just testing `selectionDirection` is truthy is enough because `DIRECTIONS.RIGHT === true` but I'd still prefer if it were tested explicitly because that's really not intuitive when you read it.

```suggestion
                    const isDirectionRight = getCursorDirection(selection.anchorNode, 0, selection.focusNode, 0) === DIRECTIONS.RIGHT;
                    if (
                        closestElement(selection.anchorNode, 'a') === link &&
                        closestElement(selection.focusNode, 'a') === link
                    ) {
                        [anchorNode, focusNode] = isDirectionRight